boost @1.76: set toolset for intel b2 invocation

scheibelp opened the pull request:
When building boost 1.76 with `%intel`, I had to set `toolset=` for the `b2` invocation or boost would choose the toolset itself (in my case gcc).

scheibelp committed:
boost @1.76: set toolset for intel b2 invocation

haampie commented on var/spack/repos/builtin/packages/boost/package.py:
```diff
@@ -446,7 +446,7 @@ def determine_b2_options(self, spec, options):
         ])
 
-        if not spec.satisfies('%intel'):
+        if (not spec.satisfies('%intel')) or spec.satisfies('@1.76:'):
```
```suggestion
        if not spec.satisfies('@:1.75 %intel')):
```
Is maybe a bit more concise? Did you check only 1.76 or also older versions? 

For reference this line was added in 2016: https://github.com/spack/spack/pull/1816#issuecomment-248991860, there's been enough releases since...

scheibelp replied:
> Did you check only 1.76 or also older versions?

It might go back farther, but I've only checked for `1.76`

scheibelp committed:
simplify constraint and add explanatory comment

scheibelp committed:
typo/more clarification

scheibelp commented:
Hi @haampie I have updated the constraint and added an explanatory comment. Do you think you'd have a chance to re-review this in the next couple days?
